### PR TITLE
remove a faulty dot

### LINKED ISSUE #58195: [BUG] network.managed add to all bridge interfaces in /etc/network/interfaces a line with a dot
**Description**
After running state.apply with an network.managed state, each br interface in /etc/network/interfaces get a line with only a dot (".") .

**Setup**
states/test.sls
```
linux_interface_eno1:
  network.managed:
  - name: eno1
  - enabled: True
  - type: eth
  - proto: manual
```

**Steps to Reproduce the behavior**
cat /etc/network/interfaces
```
auto lo
iface lo inet loopback
auto eno1
iface eno1 inet manual
auto enp3s0
iface enp3s0 inet manual
auto br-dsl
iface br-dsl inet static
    address 192.168.179.17
    netmask 255.255.255.0
    gateway 192.168.179.1
    bridge_ports eno1
```
salt-call -l debug state.apply test
```
...
[DEBUG   ] Rendered data from file: /var/cache/salt/minion/files/base/test.sls:

linux_interface_eno1:
  network.managed:
  - name: eno1
  - enabled: True
  - type: eth
  - proto: manual
[DEBUG   ] Results of YAML rendering: 
OrderedDict([('linux_interface_eno1', OrderedDict([('network.managed', [OrderedDict([('name', 'eno1')]), OrderedDict([('enabled', True)]), OrderedDict([('type', 'eth')]), OrderedDict([('proto', 'manual')])])]))])
[PROFILE ] Time (in seconds) to render '/var/cache/salt/minion/files/base/test.sls' using 'yaml' renderer: 0.0005314350128173828
[DEBUG   ] LazyLoaded config.option
[DEBUG   ] LazyLoaded ip.get_interface
[DEBUG   ] LazyLoaded network.managed
[INFO    ] Running state [eno1] at time 22:58:19.120690
[INFO    ] Executing state network.managed for [eno1]
[INFO    ] Interface eno1 is up to date.
[INFO    ] Completed state [eno1] at time 22:58:19.165807 (duration_in_ms=45.118)
[DEBUG   ] File /var/cache/salt/minion/accumulator/139801726516416 does not exist, no need to cleanup
[DEBUG   ] LazyLoaded state.check_result
[DEBUG   ] LazyLoaded highstate.output
local:

Summary for local
------------
Succeeded: 1
Failed:    0
------------
Total states run:     1
Total run time:  45.118 ms
```

cat /etc/network/interfaces
```
auto lo
iface lo inet loopback
auto eno1
iface eno1 inet manual
auto enp3s0
iface enp3s0 inet manual
auto br-dsl
iface br-dsl inet static
    address 192.168.179.17
    netmask 255.255.255.0
    gateway 192.168.179.1
.
    bridge_ports eno1
```

**Expected behavior**
If no changes, nothing should changed. 

**Versions Report**
```
Salt Version:
           Salt: 3001.1
 
Dependency Versions:
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: 2.7.3
      docker-py: Not Installed
          gitdb: Not Installed
      gitpython: Not Installed
         Jinja2: 2.10
        libgit2: Not Installed
       M2Crypto: Not Installed
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.5.6
   mysql-python: Not Installed
      pycparser: Not Installed
       pycrypto: 2.6.1
   pycryptodome: 3.6.1
         pygit2: Not Installed
         Python: 3.7.3 (default, Jul 25 2020, 13:03:44)
   python-gnupg: Not Installed
         PyYAML: 3.13
          PyZMQ: 17.1.2
          smmap: Not Installed
        timelib: Not Installed
        Tornado: 4.5.3
            ZMQ: 4.3.1
 
System Versions:
           dist: debian 10 buster
         locale: UTF-8
        machine: x86_64
        release: 4.19.0-10-amd64
         system: Linux
        version: Debian GNU/Linux 10 buster
```

**additional info**
If you have multiple bridge interfaces, all bridge interfaces are modified.

With strace, I saw, that the file is modified between this lines:
```
[INFO    ] Executing state network.managed for [eno1]
[INFO    ] Interface eno1 is up to date.
```

### PR DESCRIPTION
**Pull request by whitx**
remove a dot that cause a bug while creating bridge interfaces see #58195

### What does this PR do?

Fix a bug

### What issues does this PR fix or reference?
Fixes: #58195 

### Changelog
Remove unnecessary dot in template that cause the bridge interface to fail on debian. Fixes #58195 

### Commits signed with GPG?
No

Please review [Salt's Contributing Guide](https://docs.saltproject.io/en/master/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

### CODE REVIEW

**Review by Ch3LL — CHANGES_REQUESTED** [review]
thanks for tracking this down. Would you be willing to add a test and changelog entry for this fix so we don't miss this next time?

**Comment by Ch3LL** [comment]
I'm not seeing the changelog entry. Since the debian_ip code uses the debian_eth.jinja you should be able to add a test in `tests/unit/modules/test_debian_ip.py` to ensure this file is accurately rendered.  The `get_interface` method renders it. You can write a test that calls this method and assures the returned data is what you would expect.

**Comment by whitx** [comment]
Added changelog and unit test

**Comment by Ch3LL** [comment]
Thanks for adding the test, I really appreciate it. Looks like the debian_ip tests are failing, so that will just need to be cleaned up for this PR.

**Comment by Ch3LL** [comment]
I'm still seeing some failures on most OSs.

**Review by Ch3LL — DISMISSED** [review]

**Comment by whitx** [comment]
alright we should be good had to spend some time on this test to be right :sweat_smile:

**Comment by bdrung** [comment]
Can you rebase your merge request on master and squash your commits into one commit (with a nice commit message)? This bug was reported in Debian (https://bugs.debian.org/998655) and I want to pick that fix.

**Comment by bdrung** [comment]
Thanks for squashing. "removeing" contains a spelling mistake, but you should use imperative in the commit title anyway ("Remove faulty[...]"). See https://chris.beams.io/posts/git-commit/ for a reasoning.

File: tests/unit/modules/test_debian_ip.py [hunk 1]
```diff
@@ -561,6 +561,41 @@
...
+            'return': [
+                'auto br0\n',
+                'iface br0 inet static\n',
+                '    address 192.168.4.10\n',
+                '    netmask 255.255.255.0\n',
+                '    gateway 192.168.4.1\n',
+                '    bridge_ports eth1\n',
+                '\n']},
+
+
```
Comment (bdrung) [inline tests/unit/modules/test_debian_ip.py:598]: Empty lines with spaces.

Comment (whitx) [reply tests/unit/modules/test_debian_ip.py]: fixed and the commit message as well :)

**Review by Ch3LL — APPROVED** [review]

**Pull request merged**